test/unit/ajax.js

```diff
@@ -1408,6 +1408,98 @@ QUnit.module( "ajax", {
 		};
 	} );
 
+	ajaxTest( "jQuery.ajax() - responseURL", 13, function( assert ) {
+
+		// This function generates complete
+		// absolute URL without any relative path components
+		// Example:
+		// Input: http://localhost:9876/base/test/data/testinit.js/../../../test/data/mock.php?action=responseURL&155821644568982930
+		// Output: http://localhost:9876/base/test/data/mock.php?action=responseURL&155821644568982930
+		function removeRelativePathComponents( url ) {
+			var reg = RegExp( "(\\.\\.\\/)+" ), match;
+			match = reg.exec( url );
+			var count = ( match[ 0 ].match( /\.\.\//g ) || [] ).length, cleaned;
+			var splits = url.substring( 0, match.index ).split( "/" );
+			cleaned = splits.slice( 0, splits.length - count - 1 ).join( "/" );
+			url = cleaned + url.substring( match.index + match[ 0 ].length - 1 );
+			if ( ( url.match( /\.\.\//g ) || [] ).length !== 0 ) {
+				url = removeRelativePathComponents( url );
+			}
+			return url;
+		}
+
+		var successURL = removeRelativePathComponents( url( "mock.php?action=responseURL" ) ),
```

Comment:
You may use this normalization for _comparison_ but not when triggering a request. The `url` function logic is needed for the two ways we run unit tests (PHP & Node.js-based). You have to use it unaltered when you pass the URL to jQuery but you may normalize the URL _in assertions_, when comparing it to the desired one if it's necessary.